### ISSUE TITLE
Test dune release mode

### ISSUE THREAD

**Issue by samoht** [ocurrent/ocaml-ci#267]
There are still some quirks which can be found only when compiling the project in dune's release mode (instead of the default dev mode). For instance, some symlinks are present only in dev mode (and pass the CI steps) but can break in release mode (and be caught by `dune-release`)

/cc @hannesm who had the issue recently.

**Comment by hannesm** [comment]
yes, I had it with https://github.com/mirage/hacl/commit/76ac59fb5ca9c8cc73b298316e3f40946fac70a5 -- where `dune-release distrib` fails, the fix was https://github.com/mirage/hacl/commit/21416c913ae021a94cd534d5742bb30401f7ac82
